tools/getrealdeps.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+# getrealdeps for Chromebrew
+# Author: Satadru Pramanik (satmandu) satadru at gmail dot com
+# set -x
+pkg="${1}"
+
+RED='\e[1;91m';    # Use Light Red for errors.
+YELLOW='\e[1;33m'; # Use Yellow for informational messages.
+GREEN='\e[1;32m';  # Use Green for success messages.
+BLUE='\e[1;34m';   # Use Blue for intrafunction messages.
+GRAY='\e[0;37m';   # Use Gray for program output.
+MAGENTA='\e[1;35m';
+RESET='\e[0m'
+
+# Simplify colors and print errors to stderr (2).
+echo_error() { echo -e "${RED}${*}${RESET}" >&2; }
+echo_info() { echo -e "${YELLOW}${*}${RESET}" >&1; }
+echo_success() { echo -e "${GREEN}${*}${RESET}" >&1; }
+echo_intra() { echo -e "${BLUE}${*}${RESET}" >&1; }
+echo_out() { echo -e "${GRAY}${*}${RESET}" >&1; }
+echo_other() { echo -e "${MAGENTA}${*}${RESET}" >&2; }
+
+[[ ! -e '/usr/local/bin/grep' ]] && crew install grep
+GREP=/usr/local/bin/grep
+CREW_LIB_PREFIX=$(crew const | $GREP CREW_LIB_PREFIX | awk -F = '{print $2}')
+
+# Which packages have a needed library in CREW_LIB_PREFIX
+# This is a subset of what crew whatprovides gives
+whatprovidesfxn() {
+  pkgdepslcl="${1}"
+  filelcl=$(cd /usr/local/etc/crew/meta/ ; $GREP --exclude "${pkg}.filelist" "$pkgdepslcl" *.filelist | $GREP "$CREW_LIB_PREFIX")
+  packagelcl=$(echo "$filelcl" | \
+  sed 's/.filelist.*//g' | sed 's:.*/::' | awk '!x[$0]++' | sed s/://g)
+  echo "$packagelcl"
+}
+
+# What files does a package provide
+crewfilesfxn() {
+ pkgname="${1}"
+ files=$(cat /usr/local/etc/crew/meta/"${pkgname}".filelist)
+ echo "$files"
+}
+
+pkgfiles=$(crewfilesfxn "${pkg}")
+# Use readelf to determine library dependencie, as
+# this doesn't almost run a program like using ldd would.
+rm -rf /tmp/deps/"${pkg}"
+pkgdeps=$(unset lines ; for i in $pkgfiles ; \
+do \
+[[ $i = *.c ]] && continue
+[[ $i = *.gz ]] && continue
+[[ $i = *.h ]] && continue
+[[ $i = *.json ]] && continue
+[[ $i = *.py* ]] && continue
+[[ $i = *_tkinter* ]] && continue # Carveout for Python3
+[[ $i = *.txt ]] && continue
+[[ $i = *.xml ]] && continue
+[[ $i = *.zip ]] && continue
+mkdir -p /tmp/deps/"${pkg}"/
+lines+=$(echo ; readelf -d "$i" 2>/dev/null | $GREP NEEDED | awk '{print $5}' \
+| sed 's/\[//g' | sed 's/\]//g' | awk '!x[$0]++' | tee /tmp/deps/"${pkg}"/$(basename "$i") ; echo ) ; \
+done ; \
+echo "$lines" | tr " " "\n" \
+| awk 'NF' | awk '!x[$0]++')
+
+# Remove duplicates.
+pkgdeps=$(echo "$pkgdeps" | tr " " "\n" |  awk '!x[$0]++')
+
+# Figure out which Chromebrew packages provide the relevant deps.
+pkgdeps=$(unset lines; for j in $pkgdeps ; \
+do lines+=$(echo ; whatprovidesfxn "$j"); done ; \
+echo "$lines" | tr " " "\n" | awk '!x[$0]++')
```

Comment:
I am not following😅

Reply:
The whatprovides fxn gives us the packages which provide those files, then we clean up the output.

`awk '!x[$0]++'` removes duplicates.